File: mobile/lib/models/channel.dart
class Channel {
  final String id;
  final String name;
  final String group;
  final String? logoUrl;
  final String streamUrl;
  final bool isLive;

  const Channel({
    required this.id,
    required this.name,
    required this.group,
    this.logoUrl,
    required this.streamUrl,
    required this.isLive,
  });

  factory Channel.fromJson(Map<String, dynamic> json) {
    return Channel(
      id: json['id'] as String,
      name: json['name'] as String,
      group: json['group'] as String,
      logoUrl: json['logo_url'] as String?,
      streamUrl: json['stream_url'] as String,
      isLive: json['is_live'] as bool,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'group': group,
      'logo_url': logoUrl,
      'stream_url': streamUrl,
      'is_live': isLive,
    };
  }
}

File: backend/src/models/channel.rs
use serde::{Deserialize, Serialize};

/// Represents a single IPTV channel from an M3U playlist.
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Channel {
    /// Unique identifier derived from the stream URL hash.
    pub id: String,
    /// Display name of the channel.
    pub name: String,
    /// Group or category the channel belongs to.
    pub group: String,
    /// Optional URL to the channel logo image.
    pub logo_url: Option<String>,
    /// The HLS/MPEG-TS stream URL.
    pub stream_url: String,
    /// Whether the channel is currently reachable.
    pub is_live: bool,
    /// Optional EPG identifier used to match against XMLTV programme data.
    #[serde(skip_serializing_if = "Option::is_none")]
    pub tvg_id: Option<String>,
}
